weblate/addons/models.py

```diff
@@ -42,24 +42,42 @@
 
 
 class AddonQuerySet(models.QuerySet):
+    def filter_for_execution(self, component=None, project=None):
+        queryset = self.filter_sitewide()
+        # Include by component or project
+        if component:
+            queryset |= self.filter_component(component)
+        if project:
+            queryset |= self.filter_project(project)
+        return queryset.distinct()
+
     def filter_component(self, component):
         return self.prefetch_related("event_set").filter(
-            (Q(component=component) & Q(project_scope=False))
-            | (Q(component__project=component.project) & Q(project_scope=True))
-            | (Q(component__linked_component=component) & Q(repo_scope=True))
-            | (Q(component=component.linked_component) & Q(repo_scope=True))
+            Q(component=component)
+            | Q(project=component.project)
+            | Q(component__project=component.project)
+        )
```

Comment:
I'm still uncertain here. Do we want to list elsewhere installed add-ons on the component site? If no, it should be just `component=component` here. If yes, the repo_scope logic should stay.

Reply:
After thinking through, It would be more intuitive to not list elsewhere installed add-ons on the component site.

The AddonQuerySet would look like this

```
class AddonQuerySet(models.QuerySet):
    def filter_for_execution(self, component):
        return self.prefetch_related("event_set").filter(
            (Q(component=component) & Q(project_scope=False))
            | (Q(component__project=component.project) & Q(project_scope=True))
            | (Q(component__linked_component=component) & Q(repo_scope=True))
            | (Q(component=component.linked_component) & Q(repo_scope=True))
        )

    def filter_component(self, component):
        return self.prefetch_related("event_set").filter(component=component)

    def filter_project(self, project):
        return self.prefetch_related("event_set").filter(project=project)

    def filter_sitewide(self):
        return self.prefetch_related("event_set").filter(
            component__isnull=True, project__isnull=True
        )

    def filter_event(self, component, event):
        return component.addons_cache[event]
 ``` 
 
 wdyt?

Reply:
I agree. Just your `filter_for_execution` misses site-wide addons, so additional `| Q(component__isnull=True) & Q(project__isnull=True) is needed there.